tests/SprykerTest/Glue/AppPaymentBackendApi/Plugin/PaymentConfirmDisconnectionRequestValidatorPluginTest.php

```diff
@@ -0,0 +1,192 @@
+use Spryker\Glue\AppPaymentBackendApi\Plugin\GlueApplication\PaymentConfirmDisconnectionRequestValidatorPlugin;
+use SprykerTest\Glue\AppPaymentBackendApi\AppPaymentBackendApiPluginTester;
+
+/**
+ * Auto-generated group annotations
+ *
+ * @group SprykerTest
+ * @group Glue
+ * @group AppPaymentBackendApi
+ * @group Plugin
+ * @group PaymentConfirmDisconnectionRequestValidatorPluginTest
+ * Add your own group annotations below this line
+ */
+class PaymentConfirmDisconnectionRequestValidatorPluginTest extends Unit
+{
+    /**
+     * @var string
+     */
+    protected const TENANT_IDENTIFIER = 'tenant-identifier';
+
+    /**
+     * @see \Spryker\Glue\AppKernel\Plugin\GlueApplication\AbstractConfirmDisconnectionRequestValidatorPlugin::HEADER_CONFIRMATION_STATUS
+     *
+     * @var string
+     */
+    protected const HEADER_CONFIRMATION_STATUS = 'x-confirmation-status';
+
+    protected AppPaymentBackendApiPluginTester $tester;
+
+    public function testPaymentConfirmDisconnectionRequestValidatorPluginReturnsErrorIfThereIsNoTenantIdentifierGivenInTheRequest(): void
+    {
+        // Arrange
+        $this->tester->setDependency(AppPaymentBackendApiDependencyProvider::FACADE_TRANSLATOR, $this->getTranslatorFacadeMock());
+
+        $paymentConfirmDisconnectionRequestValidatorPlugin = new PaymentConfirmDisconnectionRequestValidatorPlugin();
+
+        // Act
+        $glueRequestValidationTransfer = $paymentConfirmDisconnectionRequestValidatorPlugin
+            ->validate(new GlueRequestTransfer());
+
+        // Assert
+        $this->assertFalse($glueRequestValidationTransfer->getIsValid());
+        $this->assertCount(1, $glueRequestValidationTransfer->getErrors());
+        $this->assertSame(
+            AppPaymentBackendApiConfig::ERROR_CODE_PAYMENT_DISCONNECTION_TENANT_IDENTIFIER_MISSING,
+            $glueRequestValidationTransfer->getErrors()[0]->getCode(),
+        );
+    }
+
+    public function testPaymentConfirmDisconnectionRequestValidatorPluginReturnsSuccessIfThereAreNoPaymentsForTenantIdentifierGivenInTheRequest(): void
+    {
+        // Arrange
+        $this->tester->setDependency(AppPaymentBackendApiDependencyProvider::FACADE_TRANSLATOR, $this->getTranslatorFacadeMock());
+
+        $paymentConfirmDisconnectionRequestValidatorPlugin = new PaymentConfirmDisconnectionRequestValidatorPlugin();
+        $glueRequestTransfer = new GlueRequestTransfer();
+        $glueRequestTransfer->setMeta([
+            GlueRequestPaymentMapper::HEADER_TENANT_IDENTIFIER => [static::TENANT_IDENTIFIER],
+        ]);
+
+        // Act
+        $glueRequestValidationTransfer = $paymentConfirmDisconnectionRequestValidatorPlugin
+            ->validate($glueRequestTransfer);
+
+        // Assert
+        $this->assertTrue($glueRequestValidationTransfer->getIsValid());
+    }
+
+    public function testPaymentConfirmDisconnectionRequestValidatorPluginReturnsErrorIfThereArePaymentsForTenantIdentifierGivenInTheRequest(): void
```

Comment:
Why should payment be in the request? You probably mean when open payment for tenant exists.

Reply:
Rephrased.